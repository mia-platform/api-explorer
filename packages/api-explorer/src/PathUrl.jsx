import {Button, Tag} from 'antd'

const React = require('react');
const PropTypes = require('prop-types');
const classNames = require('classnames');
const AuthBox = require('./AuthBox');
const Oas = require('./lib/Oas');

const { Operation } = Oas;

const extensions = require('@readme/oas-extensions');

function splitPath(path) {
  return path
    .split(/({.+?})/)
    .filter(Boolean)
    .map(part => {
      return { type: part.match(/[{}]/) ? 'variable' : 'text', value: part.replace(/[{}]/g, '') };
    });
}

function renderButtonTry(loading, onSubmit){
  /* <button
                className={classNames('api-try-it-out', { active: dirty })}
                type="submit"
                disabled={loading}
                onClick={onSubmit}
              > */
  return(
    
    <Button 
      disabled={loading}
      onClick={onSubmit}
      type={'primary'}
    >
      {!loading && (
        <span className="try-it-now-btn">
          <span className="fa fa-compass" />&nbsp;
          <span>Try It</span>
        </span>
      )}

      {loading && <i className="fa fa-circle-o-notch fa-spin" />}
    </Button>
  )
}
function renderOperationMethod(operation){
  const background = {
    post: '#52c41a',
    put: '#fa541c',
    get: '#1890ff',
    delete: '#f5222d'
  }
  return(
    <Tag color={background[operation.method]}>{operation.method}</Tag>
  )
  // return(
  //   <span 
  //     style={style} 
  //     className={`pg-type-big pg-type type-${operation.method}`}
  //   >{operation.method}</span>
  // )
}
function renderUrl(oas, operation){
  return(
    oas.servers &&
      oas.servers.length > 0 && (
        <span className="definition-url">
          <span className="url">{oas.url()}</span>
          {splitPath(operation.path).map(part => (
            <span key={part.value} className={`api-${part.type}`}>
              {part.value}
            </span>
          ))}
        </span>)
  )
}
function PathUrl({
  oas,
  operation,
  authInputRef,
  loading,
  dirty,
  onChange,
  showAuthBox,
  needsAuth,
  toggleAuth,
  onSubmit,
  oauth,
  auth,
}) {
  const containerStyle = {
    background: '#f0f2f4',
    padding: '8px 18px',
    borderRadius: 10
  }
  return (
    <div className="api-definition-parent" style={{padding: 10}}>
      <div className="api-definition" style={containerStyle}>
        {/* <div className="api-definition-container"> */}
        <div style={{display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
  
          <div>
            {renderOperationMethod(operation)}

            {renderUrl(oas, operation)}
          </div>
          
          {oas[extensions.EXPLORER_ENABLED] && (
            <div className="api-definition-actions">
              <AuthBox
                operation={operation}
                onChange={onChange}
                onSubmit={onSubmit}
                open={showAuthBox}
                needsAuth={needsAuth}
                toggle={toggleAuth}
                authInputRef={authInputRef}
                oauth={oauth}
                auth={auth}
              />

              {renderButtonTry(loading, onSubmit)}
            </div>
          )}

        </div>
      </div>
    </div>
  );
}

PathUrl.propTypes = {
  oas: PropTypes.instanceOf(Oas).isRequired,
  operation: PropTypes.instanceOf(Operation).isRequired,
  authInputRef: PropTypes.func,
  dirty: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  toggleAuth: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  showAuthBox: PropTypes.bool,
  needsAuth: PropTypes.bool,
  oauth: PropTypes.bool.isRequired,
  auth: PropTypes.shape({}),
};

PathUrl.defaultProps = {
  showAuthBox: false,
  needsAuth: false,
  authInputRef: () => {},
  auth: {},
};
module.exports = PathUrl;
module.exports.splitPath = splitPath;
