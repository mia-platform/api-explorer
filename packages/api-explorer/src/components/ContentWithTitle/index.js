import React from 'react'
import PropTypes from 'prop-types'

export default function ContentWithTitle({
  title, 
  subheader, 
  content, 
  showDivider, 
  theme, 
  showBorder,
  titleUpperCase,
  titleWithBorder
}) {
    const style = {
      light: {
        title: '#fff',
        borderContent: '#fff',
        divider: 'rgba(255,255,255,0.5)'
      },
      dark: {
        title: '#aeaeae',
        borderContent: '#000',
        divider: '#ddd'
      }
    }
    const colors = style[theme]
    const titleStyle = {
      fontSize: '18px', 
      color: colors.title, 
      ...titleUpperCase?{textTransform: 'uppercase', borderBottom: `1px solid ${colors.divider}`} : {},
      fontWeight: 'bold'
    }
    return(
      <div style={{display: 'grid', gridGap: '8px', gridTemplateColumns: '100%', gridTemplateRows: 'min-content'}}>
        <div style={titleStyle}>
          {title}
        </div>
        <div>{subheader}</div>
        <div style={showBorder ? {border: `2px solid ${colors.borderContent}`} : {}}>
          {content}
        </div>
        {showDivider ? <div style={{height: 1, background: colors.divider, margin: 0, marginTop: 5, opacity: 0.2}} /> : null}
      </div>
    )
}

ContentWithTitle.propTypes = {
    title: PropTypes.node,
    subheader: PropTypes.node,
    content: PropTypes.node,
    showDivider: PropTypes.bool,
    theme: PropTypes.oneOf(['light', 'dark']),
    showBorder: PropTypes.bool,
    titleUpperCase: PropTypes.bool
}
ContentWithTitle.defaultProps = {
  showDivider: true,
  theme: 'light',
  showBorder: true,
  titleUpperCase: false,
  titleWithBorder: false
}