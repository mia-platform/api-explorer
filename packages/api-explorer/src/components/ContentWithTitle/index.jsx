/* eslint-disable react/prop-types */
/* eslint-disable react/require-default-props */
import React from 'react'
import PropTypes from 'prop-types'

export default function ContentWithTitle({
  title,
  subheader,
  content,
  showDivider,
  theme,
  showBorder,
  titleUpperCase,
}) {
    const style = {
      light: {
        title: '#fff',
        borderContent: '#fff',
        divider: 'rgba(255,255,255,0.5)'
      },
      dark: {
        title: '#aeaeae',
        borderContent: '#000',
        divider: '#ddd'
      }
    }
    const colors = style[theme]
    const titleStyle = {
      fontSize: '18px',
      color: colors.title,
      ...titleUpperCase?{textTransform: 'uppercase', borderBottom: `1px solid ${colors.divider}`} : {},
      fontWeight: 'bold'
    }
    return(
      <div style={{display: 'grid', gridGap: '8px', gridTemplateColumns: '100%', gridTemplateRows: 'min-content', minWidth: 0}}>
        {
          title ?
            <div style={titleStyle}>
              {title}
            </div> : null
        }
        <div>{subheader}</div>
        <div style={showBorder ? {border: `2px solid ${colors.borderContent}`} : {}}>
          {content}
        </div>
        {showDivider ? <div style={{height: 1, background: colors.divider, margin: 0, marginTop: 5, opacity: 0.2}} /> : null}
      </div>
    )
}

ContentWithTitle.propTypes = {
    title: PropTypes.node,
    subheader: PropTypes.node,
    content: PropTypes.node,
    showDivider: PropTypes.bool,
    theme: PropTypes.oneOf(['light', 'dark']),
    showBorder: PropTypes.bool,
    titleUpperCase: PropTypes.bool
}
ContentWithTitle.defaultProps = {
  showDivider: true,
  theme: 'light',
  showBorder: true,
  titleUpperCase: false,
}
