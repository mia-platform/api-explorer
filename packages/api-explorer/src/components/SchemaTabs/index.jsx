import React, { Component } from 'react'
import PropTypes from 'prop-types'
import get from 'lodash.get'
import jsf from 'json-schema-faker'
import refParser from '@apidevtools/json-schema-ref-parser'
import {FormattedMessage} from 'react-intl';
import {omit} from 'ramda'
import {Alert} from 'antd'

import parametersToJsonSchema from '../../lib/parameters-to-json-schema'
import BlockWithTab from '../BlockWithTab'
import colors from '../../colors'
import JsonViewer from "../JsonViewer";
import resolveRootRef from "../../JsonForm/resolveRootRef";
import Select from "../Select";

const styleList = {
  fontSize: '18px',
  color: colors.schemaTabSelectedItem,
  textTransform: 'uppercase',
  borderBottom: `1px solid ${colors.schemaTabListBorder}`,
  fontWeight: 'bold',
  background: 'none',
  padding: 0
}
const styleSelectedItem = {
  color: colors.schemaTabSelectedItem,
  background: 'none',
  borderBottom: `3px solid ${colors.schemaTabSelectedItem}`
}
const styleLink = {
  color: 'inherit'
}
const styleItem = {
  borderBottom: `3px solid ${colors.schemaTabItemBorder}`,
  color: colors.schemaTabItemColor
}

const EXAMPLE = 'example'
const REQUEST = 'request'
const RESPONSE = 'response'

function renderMissingSchema(nameSchema) {
  return (
    <div style={{padding: 10, background: colors.schemaTabMissingSchemaBackground}}>
      <FormattedMessage id={`schemaTabs.missing.${nameSchema.toLowerCase()}`} defaultMessage={`${nameSchema} schema not set`} />
    </div>
  )
}

jsf.option({
  failOnInvalidTypes: false,
  failOnInvalidFormat: false,
  useDefaultValue: true,
})

const styles = {
  responseSchemaWrapper: {
    display: 'flex',
    flexDirection: 'column'
  },
  responseSchemaSelect: {
    margin: '0 0 8px auto',
  }
}

export default class SchemaTabs extends Component {
  constructor(props) {
    super(props)
    this.state = {
      selected: EXAMPLE,
      selectedStatus: undefined,
      schema: undefined
    }
  }

  static getDerivedStateFromProps(props, state) {
    const {operation} = props
    if (operation && operation.responses && !state.selectedStatus) {
      return {selectedStatus: Object.keys(operation.responses)[0]}
    }
    return null
  }

  componentDidMount() {
    const { operation, oas } = this.props
    const schema = get(parametersToJsonSchema(operation, oas), '[0].schema', {})
    refParser.dereference(resolveRootRef(schema), (err, schemaResolved) => {
      if (!err) {
        this.setState({schema: omit(['components'], schemaResolved)})
        return
      }
      this.setState({schema: omit(['components'], schema)})
    })
  }

  renderSchemaExample() {
    const {schema} = this.state
    let example = get(schema, EXAMPLE)

    if (schema.type === 'array') {
      example = get(schema.items, EXAMPLE)
    }

    try {
      return (
        <JsonViewer
          missingMessage={'schemaTabs.missing.example'}
          schema={example || jsf.generate(schema)}
          key={'json-viewer-example'}
        />
      )
    } catch (error) {
      return <Alert type={'error'} message={error.message} />
    }
  }

  renderResponseSchema() {
    const {operation} = this.props
    const {selectedStatus} = this.state
    return (
      <div style={styles.responseSchemaWrapper}>
        <Select
          firstActiveValue={selectedStatus}
          options={Object.keys(operation.responses)}
          onChange={value => this.setState({selectedStatus: value})}
          value={this.state.selectedStatus}
          style={styles.responseSchemaSelect}
        />
        <JsonViewer missingMessage={'schemaTabs.missing.response'} schema={operation.responses[selectedStatus]} />
      </div>
    )
  }

  renderRequestSchema() {
    const {schema} = this.state
    return (
      <JsonViewer
        key={'json-viewer-request'}
        missingMessage={'schemaTabs.missing.example'}
        schema={schema}
      />
    )
  }

  renderSchema () {
    const {operation} = this.props
    const {selected, schema} = this.state
    const hasSchema = schema && Object.keys(schema).length > 0
    const hasResponses = operation && operation.responses && Object.keys(operation.responses).length > 0

    const selectedType = () => {
      switch (selected) {
        case REQUEST: {
          return hasSchema ? this.renderRequestSchema() : renderMissingSchema(REQUEST)
        }
        case RESPONSE: {
          return hasResponses ? this.renderResponseSchema() : renderMissingSchema(RESPONSE)
        }
        case EXAMPLE: {
          return hasSchema ? this.renderSchemaExample() : renderMissingSchema(EXAMPLE)
        }
        default: {
          return null
        }
      }
    }

    return (
      <div style={{paddingTop: 10}}>
        {selectedType()}
      </div>
    )
  }

  render() {
    const {selected} = this.state
    return (
      <BlockWithTab
        items={[
          { value: EXAMPLE, label: <FormattedMessage id='schemaTabs.label.example' defaultMessage='Example' /> },
          { value: REQUEST, label: <FormattedMessage id='schemaTabs.label.request' defaultMessage='Request' /> },
          { value: RESPONSE, label: <FormattedMessage id='schemaTabs.label.response' defaultMessage='Response' />}
        ]}
        selected={selected}
        styleList={styleList}
        styleSelectedItem={styleSelectedItem}
        styleLink={styleLink}
        styleItem={styleItem}
        onClick={(item) => this.setState({selected: item})}
      >
        {this.renderSchema()}
      </BlockWithTab>
    )
  }
}

SchemaTabs.propTypes = {
  operation: PropTypes.object, // eslint-disable-line react/require-default-props
  oas: PropTypes.object // eslint-disable-line react/require-default-props
}
