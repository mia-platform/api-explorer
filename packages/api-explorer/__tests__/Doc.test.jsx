import { mountWithIntl } from 'enzyme-react-intl';
import { IntlProvider, FormattedMessage } from 'react-intl';
import {shallow} from 'enzyme'
import {Button} from 'antd'

import ContentWithTitle from '../src/components/ContentWithTitle'
import SchemaTabs from '../src/components/SchemaTabs'
import PathUrl from '../src/PathUrl'

const extensions = require('@mia-platform/oas-extensions');
const { Request, Response } = require('node-fetch');

global.Request = Request;

const React = require('react');

const Doc = require('../src/Doc');
const oas = require('./fixtures/petstore/circular-oas');
const oasWithSlashes = require('./fixtures/petstore/oas')

const props = {
  doc: {
    title: 'Title',
    slug: 'slug',
    type: 'endpoint',
    swagger: { path: '/pet/{petId}' },
    api: { method: 'get' },
    formData: { path: { petId: '1' }, auth: { api_key: '' } },
    onSubmit: () => {},
  },
  oas,
  setLanguage: () => {},
  language: 'node',
  suggestedEdits: false,
  oauth: false,
  auth: {},
  tryItMetrics: () => {},
};

function assertDocElements(component, doc) {
  expect(component.find(`#page-${doc.slug}`).length).toBe(1);
}

test('should output correct components', (done) => {
  const wrapper = mountWithIntl(
    <IntlProvider>
      <Doc {...props} />
    </IntlProvider>
  );
  setTimeout(() => {
    wrapper.mount()
    assertDocElements(wrapper, props.doc);
    expect(wrapper.find(`div#page-${props.doc.slug}`).length).toBe(1);
    expect(wrapper.find('PathUrl').length).toBe(1);
    expect(wrapper.find('CodeSample').length).toBe(1);
    expect(wrapper.find('Params').length).toBe(1);
    expect(wrapper.find(ContentWithTitle)).toHaveLength(5);
    expect(wrapper.find(SchemaTabs)).toHaveLength(1)
    done()
  }, 0)
});

test('should work without a doc.swagger/doc.path/oas', () => {
  const doc = { title: 'title', slug: 'slug', type: 'basic' };
  const docComponent = mountWithIntl(
    <IntlProvider>
      <Doc
        doc={doc}
        setLanguage={() => {}}
        language="node"
        suggestedEdits
        oauth={false}
        tryItMetrics={() => {}}
        onAuthChange={() => {}}
        auth={{}}
      />
    </IntlProvider>
  );

  assertDocElements(docComponent, doc);
  expect(docComponent.find(`div#page-${props.doc.slug}`).length).toBe(1);
  expect(docComponent.find('ContentWithTitle').length).toBe(0);
});

test('should still display `Content` with column-style layout', () => {
  const doc = { title: 'title', slug: 'slug', type: 'basic' };
  const docComponent = mountWithIntl(
    <IntlProvider>
      <Doc
        doc={doc}
        setLanguage={() => {}}
        language="node"
        suggestedEdits
        appearance={{ referenceLayout: 'column' }}
        oauth={false}
        tryItMetrics={() => {}}
        onAuthChange={() => {}}
        auth={{}}
      />
    </IntlProvider>
  );

  assertDocElements(docComponent, doc);
  expect(docComponent.find('ContentWithTitle').length).toBe(0);
});

describe('state.dirty', () => {
  test('should default to false', () => {
    const doc = shallow(<Doc {...props} />);

    expect(doc.state('dirty')).toBe(false);
  });

  test('should switch to true on form change', () => {
    const doc = shallow(<Doc {...props} />);
    doc.instance().onChange({ a: 1 });

    expect(doc.state('dirty')).toBe(true);
  });
});

describe('onSubmit', () => {
  test('should display authentication warning if auth is required for endpoint', () => {
    jest.useFakeTimers();

    const doc = shallow(<Doc {...props} />);

    // mock authInput focus function.
    doc.instance().authInput = {focus: () => {}}
    doc.instance().onSubmit();
    expect(doc.state('showAuthBox')).toBe(true);

    jest.runAllTimers();

    expect(doc.state('needsAuth')).toBe(true);
  });

  test('should make request on Submit', () => {
    const props2 = {
      doc: {
        title: 'Title',
        slug: 'slug',
        type: 'endpoint',
        swagger: { path: '/pet' },
        api: { method: 'post' },
        formData: {
          body: {
            name: '1',
            photoUrls: ['1'],
          },
        },
        onSubmit: () => {},
      },
      oas,
      setLanguage: () => {},
      language: 'node',
      oauth: false,
      auth: { petstore_auth: 'api-key' },
    };

    const fetch = window.fetch;

    window.fetch = request => {
      expect(request.url).toContain(oas.servers[0].url);
      return Promise.resolve(
        new Response(JSON.stringify({ id: 1 }), {
          headers: { 'content-type': 'application/json' },
        }),
      );
    };

    const doc = shallow(<Doc {...props} {...props2} />);

    doc
      .instance()
      .onSubmit()
      .then(() => {
        expect(doc.state('loading')).toBe(false);
        expect(doc.state('result')).not.toEqual(null);

        window.fetch = fetch;
      });
  });

  test('should not make request to the proxy url if necessary (proxy feature has been removed)', () => {
    const proxyOas = {
      servers: [{ url: 'http://example.com' }],
      [extensions.PROXY_ENABLED]: true,
      paths: {
        '/pet/{petId}': {
          get: {
            responses: {
              default: {
                description: 'desc',
              },
            },
          },
        },
      },
    };

    const doc = shallow(<Doc {...props} oas={proxyOas} />);

    const fetch = window.fetch;

    window.fetch = request => {
      expect(request.url).toContain(`${proxyOas.servers[0].url}`);
      return Promise.resolve(new Response());
    };

    doc.instance().onSubmit();

    window.fetch = fetch;
  });

  test('should call `tryItMetrics` on success', async () => {
    let called = false;

    const doc = shallow(
      <Doc
        {...props}
        tryItMetrics={() => {
            called = true;
          }}
        auth={{ api_key: 'api-key' }}
      />
    );

    const fetch = window.fetch;
    window.fetch = () => {
      return Promise.resolve(new Response());
    };

    await doc.instance().onSubmit();
    expect(called).toBe(true);
    window.fetch = fetch;
  });
});

describe('toggleAuth', () => {
  test('toggleAuth should change state of showAuthBox', () => {
    const doc = shallow(<Doc {...props} />);

    expect(doc.state('showAuthBox')).toBe(false);

    doc.instance().toggleAuth({ preventDefault() {} });

    expect(doc.state('showAuthBox')).toBe(true);

    doc.instance().toggleAuth({ preventDefault() {} });

    expect(doc.state('showAuthBox')).toBe(false);
  });
});

describe('PathUrl', () => {
  it('should be passed auth from props if the one in state is undefined', () => {
    const doc = shallow(<Doc {...props} auth={{api_key: '123' }} />);
    const pathUrl = doc.find(PathUrl)
    expect(pathUrl.prop('auth')).toEqual({api_key: '123' })
  })
  it('should be passed auth from state if the one in state is not undefined', () => {
    const doc = shallow(<Doc {...props} auth={{api_key: '123' }} />);
    const pathUrl = doc.find(PathUrl)
    pathUrl.prop('onChange')({api_key: '456'})
    doc.update()
    expect(doc.find(PathUrl).prop('auth')).toEqual({api_key: '456' })
  })
})

describe('state.loading', () => {
  test('should default to false', () => {
    const doc = shallow(<Doc {...props} />);
    expect(doc.state('loading')).toBe(false);
  });
});

test('should output with an error message if the endpoint fails to load', () => {
  const brokenOas = {
    paths: {
      '/path': {
        post: {
          requestBody: {
            $ref: '#/components/schemas/UnknownSchema',
          },
        },
      },
    },
  };

  const doc = shallow(
    <Doc
      {...props}
      oas={brokenOas}
      doc={{
          title: 'title',
          slug: 'slug',
          type: 'endpoint',
          swagger: { path: '/path' },
          api: { method: 'post' },
        }}
    />
  );
  expect(doc.find('ErrorBoundary').length).toBe(1);
});

describe('fallbackUrl', () => {
  it('should inject servers in oas if it has no server', () => {
    const fallback = 'https://somexample.com/';
    const clonedOas = Object.assign({}, oas);
    delete clonedOas.servers;

    const doc = shallow(
      <Doc
        {...props}
        oas={clonedOas}
        fallbackUrl={fallback}
      />
    )
    expect(doc.instance().oas.servers).toEqual([{url: fallback}])
  })
})

describe('stripSlash', () => {
  it('should default to false', () => {
    const doc = shallow(<div>{<Doc {...props} />}</div>).find(Doc)
    expect(doc.prop('stripSlash')).toBe(false)
  })

  it('should strip operation path', () => {
    const doc = shallow(
      <Doc
        {...props}
        stripSlash
        doc={{
            title: 'Title',
            slug: 'slug',
            type: 'endpoint',
            swagger: { path: '/store/inventory/' },
            api: { method: 'get' },
            formData: { path: { petId: '1' }, auth: { api_key: '' } },
            onSubmit: () => {},
          }}
        oas={oasWithSlashes}
      />
    )

    const operation = doc.instance().getOperation()
    expect(operation.path).toBe('/store/inventory')
  })

  it('should not strip operation path', () => {
    const doc = shallow(
      <Doc
        {...props}
        doc={{
            title: 'Title',
            slug: 'slug',
            type: 'endpoint',
            swagger: { path: '/store/inventory/' },
            api: { method: 'get' },
            formData: { path: { petId: '1' }, auth: { api_key: '' } },
            onSubmit: () => {},
          }}
        oas={oasWithSlashes}
        stripSlash={false}
      />
    )

    const operation = doc.instance().getOperation()
    expect(operation.path).toBe('/store/inventory/')
  })
})

test('expand renderCodeAndResponse should be render correctly', () => {
  const element = mountWithIntl(<Doc {...props} />)
  expect(element.find('.expandable').find(Button).find(FormattedMessage).prop('id')).toEqual('doc.expand')
  expect(element.find('.expandable').find(Button).prop('icon')).toEqual('import')
  expect(element.find('.expandable').prop('style')).toMatchSnapshot()
  element.find('.expandable').find(Button).simulate('click')
  expect(element.find('.expandable').find(Button).find(Button).find(FormattedMessage).prop('id')).toEqual('doc.collapse')
  expect(element.find('.expandable').find(Button).find(Button).prop('icon')).toEqual('export')
  expect(element.find('.expandable').prop('style')).toMatchSnapshot()
})
