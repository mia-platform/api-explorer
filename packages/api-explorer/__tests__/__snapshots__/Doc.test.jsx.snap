// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expand renderCodeAndResponse should be render correctly 1`] = `
Object {
  "background": "rgb(51, 55, 58)",
  "border": "1px solid #0f0f0f",
}
`;

exports[`expand renderCodeAndResponse should be render correctly 2`] = `
Object {
  "background": "rgb(51, 55, 58)",
  "border": "1px solid #0f0f0f",
  "height": "100%",
  "left": 0,
  "position": "absolute",
  "width": "100%",
}
`;
